hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java

```diff
@@ -128,35 +110,17 @@ public void execute(Runnable command) {
      * @param command
      * @param delay
      * @param unit
-     * @param executionCallback
      * @return scheduledFuture
      */
-    public ScheduledFuture<?> schedule(final Runnable command, long delay, TimeUnit unit,
-                                       final ExecutionCallback executionCallback) {
-        return scheduledExecutor.schedule(new Runnable() {
-            public void run() {
-                executeInternalSafely(command, executionCallback);
-            }
-        }, delay, unit);
-    }
-
     @Override
     public ScheduledFuture<?> schedule(final Runnable command, long delay, TimeUnit unit) {
-        return scheduledExecutor.schedule(new Runnable() {
-            public void run() {
-                executeInternalSafely(command, failureLoggingExecutionCallback);
-            }
-        }, delay, unit);
+        return internalExecutor.schedule(command, delay, unit);
     }
 
     @Override
     public ScheduledFuture<?> scheduleWithRepetition(final Runnable command, long initialDelay, long period, TimeUnit unit) {
         final Runnable decoratedCommand = new SkipOnConcurrentExecutionDecorator(command);
-        return scheduledExecutor.scheduleAtFixedRate(new Runnable() {
-            public void run() {
-                executeInternalSafely(decoratedCommand, failureLoggingExecutionCallback);
-            }
-        }, initialDelay, period, unit);
+        return internalExecutor.scheduleAtFixedRate(decoratedCommand, initialDelay, period, unit);
```

Comment:
@jerrinot I have removed the decorator. I was thinking by leaving that we will make sure the match the semantics. But, after your warning, it occurred to me that decorator is useless here. 
Although I don't see this as problem,  we will end up with slightly different semantics here. 
 `scheduleAtFixedRate ` will run next task immediately in case of the previous one takes longer than the period. But `scheduleWithRepetition` on node side will wait for next period checkpoint.